plugins/policy-recommendation/policy_recommendation_job_test.py

```diff
@@ -0,0 +1,1215 @@
+    ),
+    (
+        (
+            ('antrea-test#{"podname":"perftest-a"}#5201#TCP', ""),
+            ("", ""),
+        ),
+        ('antrea-test#{"podname":"perftest-a"}#5201#TCP', "")
+    ),
+    (
+        (
+            ('antrea-test#{"podname":"perftest-a"}#5201#TCP', ""),
+            ("", 'antrea-test#{"podname":"perftest-b"}#5201#TCP'),
+        ),
+        ('antrea-test#{"podname":"perftest-a"}#5201#TCP', 'antrea-test#{"podname":"perftest-b"}#5201#TCP')
+    ),
+])
+def test_combine_network_peers(test_input, expected_peer):
+    peer1, peer2 = test_input
+    network_peer = combine_network_peers(peer1, peer2)
+    assert network_peer == expected_peer
+
+@pytest.mark.parametrize("test_input, expected_egress_rule", [
+    (
+        'wrongformat',
+        "",
+    ),
+    (
+        'antrea-test#{"podname":"perftest-a"}#5201#TCP',
+        kubernetes.client.V1NetworkPolicyEgressRule(
+            to=[
+                kubernetes.client.V1NetworkPolicyPeer(
+                    namespace_selector = kubernetes.client.V1LabelSelector(
+                        match_labels = {
+                            "name":'antrea-test'
+                        }
+                    ),
+                    pod_selector = kubernetes.client.V1LabelSelector(
+                        match_labels = {
+                            "podname":"perftest-a"
+                        }
+                    ),    
+                )
+            ],
+            ports=[
+                kubernetes.client.V1NetworkPolicyPort(
+                    port = 5201,
+                    protocol = "TCP"
+                )
+            ]
+        )
+    ),
+    (
+        '192.168.0.1#80#TCP',
+        kubernetes.client.V1NetworkPolicyEgressRule(
+            to=[
+                kubernetes.client.V1NetworkPolicyPeer(
+                    ip_block = kubernetes.client.V1IPBlock(
+                        cidr = "192.168.0.1/32",
+                    )
+                )
+            ],
+            ports=[
+                kubernetes.client.V1NetworkPolicyPort(
+                    port = 80,
+                    protocol = "TCP"
+                )
+            ]
+        )
+    ),
+    (
+        'fd12:3456:789a:1::1#80#TCP',
+        kubernetes.client.V1NetworkPolicyEgressRule(
+            to=[
+                kubernetes.client.V1NetworkPolicyPeer(
+                    ip_block = kubernetes.client.V1IPBlock(
+                        cidr = "fd12:3456:789a:1::1/128",
+                    )
+                )
+            ],
+            ports=[
+                kubernetes.client.V1NetworkPolicyPort(
+                    port = 80,
+                    protocol = "TCP"
+                )
+            ]
+        )
+    ),
+])
+def test_generate_k8s_egress_rule(test_input, expected_egress_rule):
+    if test_input == "wrongformat":
+        with pytest.raises(SystemExit) as pytest_wrapped_e:
+                generate_k8s_egress_rule(test_input)
```

Comment:
can this call raise a SystemExit error?

Reply:
Understood!

One thing we might consider for the polcy reco job is instead of having sys.exit throughout the code is to raise specific exception (we can also define exception types if needed) and then let those exception bubble up to the main function where we can use the exception type to exit with the appropriate status code.

This is more for code readability/maintanability, but will also allow us to capture precise exception types in unit tests.

We can iterate over this with another PR, for this PR it's perfectly fine to check for SystemExit